--- meson.build	2019-10-04 10:26:51.000000000 +0200
+++ meson.build.new	2019-10-04 16:36:30.209153040 +0200
@@ -23,9 +23,7 @@
 
 # create a static library from a shared one extracting all objects
 # this allows to rewrite part of it if necessary for mocking
-test_lib = \
-  static_library('test-lib',
-                 objects : spice_client_glib_lib.extract_all_objects())
+test_lib = static_library('test-lib', objects : spice_client_glib_lib.extract_all_objects())
 
 foreach src : tests_sources
   name = 'test-@0@'.format(src).split('.')[0]
