diff -Naur crun/src/libcrun/error.c crun_tainted/src/libcrun/error.c
--- crun/src/libcrun/error.c	2024-11-10 19:35:41.000000000 +0100
+++ crun_tainted/src/libcrun/error.c	2024-11-26 10:25:24.333710458 +0100
@@ -44,6 +44,15 @@
 static int log_format;
 static int output_verbosity = LIBCRUN_VERBOSITY_ERROR;
 
+/*-------------------------------------------------------------------------*/
+#include <syslog.h>
+#define KERR(format, a...)                                    \
+ do {                                                         \
+    syslog(LOG_ERR | LOG_USER, "KERR line:%d " format "\n\n", \
+    __LINE__, ## a);                                \
+    } while (0)
+/*-------------------------------------------------------------------------*/
+
 int
 libcrun_make_error (libcrun_error_t *err, int status, const char *msg, ...)
 {
diff -Naur crun/src/libcrun/linux.c crun_tainted/src/libcrun/linux.c
--- crun/src/libcrun/linux.c	2024-11-10 19:35:41.000000000 +0100
+++ crun_tainted/src/libcrun/linux.c	2024-11-26 10:25:35.581749420 +0100
@@ -105,6 +105,15 @@
 #  define FSCONFIG_CMD_CREATE 6
 #endif
 
+/*-------------------------------------------------------------------------*/
+#include <syslog.h>
+#define KERR(format, a...)                                    \
+ do {                                                         \
+    syslog(LOG_ERR | LOG_USER, "KERR line:%d " format "\n\n", \
+    __LINE__, ## a);                                \
+    } while (0)
+/*-------------------------------------------------------------------------*/
+
 struct remount_s
 {
   struct remount_s *next;
@@ -1580,6 +1589,10 @@
                                   { "/dev/tty", "c", 5, 0, 0666, 0, 0 },
                                   { "/dev/random", "c", 1, 8, 0666, 0, 0 },
                                   { "/dev/urandom", "c", 1, 9, 0666, 0, 0 },
+                                  { "/dev/fuse", "c", 10, 229, 0666, 0, 0 },
+                                  { "/dev/kvm", "c", 10, 232, 0666, 0, 0 },
+                                  { "/dev/vhost-net", "c", 10, 238, 0666, 0, 0 },
+                                  { "/dev/net/tun", "c", 10, 200, 0666, 0, 0 },
                                   {} };
 
 /* Check if the specified path is a direct child of /dev.  If it is
