diff -Naur crun/src/libcrun/error.c crun_tainted/src/libcrun/error.c
--- crun/src/libcrun/error.c	2025-05-30 08:02:09.000000000 +0200
+++ crun_tainted/src/libcrun/error.c	2025-05-30 08:24:03.637173154 +0200
@@ -44,6 +44,15 @@
 static int log_format;
 static int output_verbosity = LIBCRUN_VERBOSITY_ERROR;
 
+/*-------------------------------------------------------------------------*/
+#include <syslog.h>
+#define KERR(format, a...)                                    \
+ do {                                                         \
+    syslog(LOG_ERR | LOG_USER, "KERR line:%d " format "\n\n", \
+    __LINE__, ## a);                                \
+    } while (0)
+/*-------------------------------------------------------------------------*/
+
 int
 libcrun_make_error (libcrun_error_t *err, int status, const char *msg, ...)
 {
diff -Naur crun/src/libcrun/linux.c crun_tainted/src/libcrun/linux.c
--- crun/src/libcrun/linux.c	2025-05-30 08:02:09.000000000 +0200
+++ crun_tainted/src/libcrun/linux.c	2025-05-30 08:24:03.637173154 +0200
@@ -110,6 +110,15 @@
 #define ALL_PROPAGATIONS_NO_REC (MS_SHARED | MS_PRIVATE | MS_SLAVE | MS_UNBINDABLE)
 #define ALL_PROPAGATIONS (MS_REC | ALL_PROPAGATIONS_NO_REC)
 
+/*-------------------------------------------------------------------------*/
+#include <syslog.h>
+#define KERR(format, a...)                                    \
+ do {                                                         \
+    syslog(LOG_ERR | LOG_USER, "KERR line:%d " format "\n\n", \
+    __LINE__, ## a);                                \
+    } while (0)
+/*-------------------------------------------------------------------------*/
+
 struct remount_s
 {
   struct remount_s *next;
@@ -1571,6 +1580,10 @@
                                   { "/dev/tty", "c", 5, 0, 0666, 0, 0 },
                                   { "/dev/random", "c", 1, 8, 0666, 0, 0 },
                                   { "/dev/urandom", "c", 1, 9, 0666, 0, 0 },
+                                  { "/dev/fuse", "c", 10, 229, 0666, 0, 0 },
+                                  { "/dev/kvm", "c", 10, 232, 0666, 0, 0 },
+                                  { "/dev/vhost-net", "c", 10, 238, 0666, 0, 0 },
+                                  { "/dev/net/tun", "c", 10, 200, 0666, 0, 0 },
                                   {} };
 
 /* Check if the specified path is a direct child of /dev.  If it is
@@ -1971,9 +1984,9 @@
   if (strcmp (destination, "/dev/pts") == 0)
     {
       if (container->host_uid == 0)
-        *data = xstrdup ("newinstance,ptmxmode=0666,mode=620,gid=5");
+        *data = xstrdup ("newinstance,ptmxmode=0666,mode=666,gid=5");
       else
-        *data = xstrdup ("newinstance,ptmxmode=0666,mode=620");
+        *data = xstrdup ("newinstance,ptmxmode=0666,mode=666");
       return MS_NOEXEC | MS_NOSUID;
     }
   if (strcmp (destination, "/sys") == 0)
