diff -Naur tldk/examples/l4fwd/main.c tldk_tainted/examples/l4fwd/main.c
--- tldk/examples/l4fwd/main.c	2020-05-08 19:11:11.000000000 +0200
+++ tldk_tainted/examples/l4fwd/main.c	2020-05-12 23:10:23.666207475 +0200
@@ -21,8 +21,8 @@
 #define	MAX_RULES	0x100
 #define	MAX_TBL8	0x800
 
-#define	RX_RING_SIZE	0x400
-#define	TX_RING_SIZE	0x800
+#define	RX_RING_SIZE	0x100
+#define	TX_RING_SIZE	0x100
 
 #define	MPOOL_CACHE_SIZE	0x100
 #define	MPOOL_NB_BUF		0x20000
