diff -Naur ovs/lib/poll-loop.c ovs_tainted/lib/poll-loop.c
--- ovs/lib/poll-loop.c	2022-06-23 22:36:48.000000000 +0200
+++ ovs_tainted/lib/poll-loop.c	2022-06-26 15:31:02.962918433 +0200
@@ -64,7 +64,7 @@
     struct poll_node *node;
 
     /* Both 'fd' and 'wevent' cannot be set. */
-    ovs_assert(!fd != !wevent);
+// cloonix fd is null on vm    ovs_assert(!fd != !wevent);
 
     HMAP_FOR_EACH_WITH_HASH (node, hmap_node,
                              hash_2words(fd, (uint32_t)wevent),
@@ -108,7 +108,7 @@
     COVERAGE_INC(poll_create_node);
 
     /* Both 'fd' and 'wevent' cannot be set. */
-    ovs_assert(!fd != !wevent);
+//cloonix fd can be 0    ovs_assert(!fd != !wevent);
 
     /* Check for duplicate.  If found, "or" the events. */
     node = find_poll_node(loop, fd, wevent);
diff -Naur ovs/ofproto/ofproto-dpif.c ovs_tainted/ofproto/ofproto-dpif.c
--- ovs/ofproto/ofproto-dpif.c	2022-06-23 22:36:48.000000000 +0200
+++ ovs_tainted/ofproto/ofproto-dpif.c	2022-06-26 15:30:06.838651014 +0200
@@ -744,7 +744,7 @@
         return 0;
     }
 
-    backer_name = xasprintf("ovs-%s", type);
+    backer_name = xasprintf("%s-%s", getenv("CLOONIX_NET"), type);
 
     /* Remove any existing datapaths, since we assume we're the only
      * userspace controlling the datapath. */
