diff -Naur ovs/lib/dpif-netlink.c ovs_tainted/lib/dpif-netlink.c
--- ovs/lib/dpif-netlink.c	2024-11-10 19:48:17.000000000 +0100
+++ ovs_tainted/lib/dpif-netlink.c	2024-11-28 21:47:18.558903588 +0100
@@ -4595,13 +4595,13 @@
 dpif_netlink_init(void)
 {
     static struct ovsthread_once once = OVSTHREAD_ONCE_INITIALIZER;
-    static int error;
+    static int error = 0;
 
     if (ovsthread_once_start(&once)) {
         error = nl_lookup_genl_family(OVS_DATAPATH_FAMILY,
                                       &ovs_datapath_family);
         if (error) {
-            VLOG_INFO("Generic Netlink family '%s' does not exist. "
+            VLOG_ERR("Generic Netlink family '%s' does not exist. "
                       "The Open vSwitch kernel module is probably not loaded.",
                       OVS_DATAPATH_FAMILY);
         }
diff -Naur ovs/lib/poll-loop.c ovs_tainted/lib/poll-loop.c
--- ovs/lib/poll-loop.c	2024-11-10 19:48:17.000000000 +0100
+++ ovs_tainted/lib/poll-loop.c	2024-11-28 21:47:18.558903588 +0100
@@ -64,7 +64,7 @@
     struct poll_node *node;
 
     /* Both 'fd' and 'wevent' cannot be set. */
-    ovs_assert(!fd != !wevent);
+// cloonix fd is null on vm    ovs_assert(!fd != !wevent);
 
     HMAP_FOR_EACH_WITH_HASH (node, hmap_node,
                              hash_2words(fd, (uint32_t)wevent),
@@ -108,7 +108,7 @@
     COVERAGE_INC(poll_create_node);
 
     /* Both 'fd' and 'wevent' cannot be set. */
-    ovs_assert(!fd != !wevent);
+//cloonix fd can be 0    ovs_assert(!fd != !wevent);
 
     /* Check for duplicate.  If found, "or" the events. */
     node = find_poll_node(loop, fd, wevent);
diff -Naur ovs/ofproto/ofproto-dpif.c ovs_tainted/ofproto/ofproto-dpif.c
--- ovs/ofproto/ofproto-dpif.c	2024-11-10 19:48:17.000000000 +0100
+++ ovs_tainted/ofproto/ofproto-dpif.c	2024-11-28 21:47:18.558903588 +0100
@@ -747,7 +747,7 @@
         return 0;
     }
 
-    backer_name = xasprintf("ovs-%s", type);
+    backer_name = xasprintf("%s-%s", getenv("CLOONIX_NET"), type);
 
     /* Remove any existing datapaths, since we assume we're the only
      * userspace controlling the datapath. */
