diff -Naur ovs/acinclude.m4 ovs_tainted/acinclude.m4
--- ovs/acinclude.m4	2021-08-29 17:22:49.000000000 +0200
+++ ovs_tainted/acinclude.m4	2021-08-29 17:30:55.549392714 +0200
@@ -434,6 +434,10 @@
           esac
     esac
 
+    DPDK_INCLUDE="$DPDK_INCLUDES"
+    DPDK_LIB="$DPDK_LIBS"
+
+
     ovs_save_CFLAGS="$CFLAGS"
     ovs_save_LDFLAGS="$LDFLAGS"
     CFLAGS="$CFLAGS $DPDK_INCLUDE"
@@ -476,7 +480,7 @@
     OVS_FIND_DEPENDENCY([dlopen], [dl], [libdl])
 
     AC_MSG_CHECKING([whether linking with dpdk works])
-    LIBS="$DPDK_LIB $LIBS"
+    LIBS="$DPDK_LIB"
     AC_LINK_IFELSE(
       [AC_LANG_PROGRAM([#include <rte_config.h>
                         #include <rte_eal.h>],
@@ -524,6 +528,8 @@
     AC_DEFINE([DPDK_NETDEV], [1], [System uses the DPDK module.])
   fi
 
+  DPDK_vswitchd_LDFLAGS=$DPDK_LIB
+
   AM_CONDITIONAL([DPDK_NETDEV], test "$DPDKLIB_FOUND" = true)
 ])
 
diff -Naur ovs/ofproto/ofproto-dpif.c ovs_tainted/ofproto/ofproto-dpif.c
--- ovs/ofproto/ofproto-dpif.c	2021-08-29 17:22:49.000000000 +0200
+++ ovs_tainted/ofproto/ofproto-dpif.c	2021-08-29 17:31:04.401434418 +0200
@@ -748,7 +748,7 @@
         return 0;
     }
 
-    backer_name = xasprintf("ovs-%s", type);
+    backer_name = xasprintf("%s-%s", getenv("CLOONIX_NET"), type);
 
     /* Remove any existing datapaths, since we assume we're the only
      * userspace controlling the datapath. */
 
