diff -Naur ovs/acinclude.m4 ovs_tainted/acinclude.m4
--- ovs/acinclude.m4	2021-11-27 15:39:37.000000000 +0100
+++ ovs_tainted/acinclude.m4	2021-12-03 07:49:01.129772606 +0100
@@ -434,6 +434,10 @@
           esac
     esac
 
+    DPDK_INCLUDE="$DPDK_INCLUDES"
+    DPDK_LIB="$DPDK_LIBS"
+
+
     ovs_save_CFLAGS="$CFLAGS"
     ovs_save_LDFLAGS="$LDFLAGS"
     CFLAGS="$CFLAGS $DPDK_INCLUDE"
@@ -476,7 +480,7 @@
     OVS_FIND_DEPENDENCY([dlopen], [dl], [libdl])
 
     AC_MSG_CHECKING([whether linking with dpdk works])
-    LIBS="$DPDK_LIB $LIBS"
+    LIBS="$DPDK_LIB"
     AC_LINK_IFELSE(
       [AC_LANG_PROGRAM([#include <rte_config.h>
                         #include <rte_eal.h>],
@@ -524,6 +528,8 @@
     AC_DEFINE([DPDK_NETDEV], [1], [System uses the DPDK module.])
   fi
 
+  DPDK_vswitchd_LDFLAGS=$DPDK_LIB
+
   AM_CONDITIONAL([DPDK_NETDEV], test "$DPDKLIB_FOUND" = true)
 ])
 
diff -Naur ovs/lib/netdev-dpdk.c ovs_tainted/lib/netdev-dpdk.c
--- ovs/lib/netdev-dpdk.c	2021-11-27 15:39:37.000000000 +0100
+++ ovs_tainted/lib/netdev-dpdk.c	2021-12-03 07:49:01.129772606 +0100
@@ -38,6 +38,7 @@
 #include <rte_pci.h>
 #include <rte_version.h>
 #include <rte_vhost.h>
+#include <ethdev_driver.h>
 
 #include "cmap.h"
 #include "coverage.h"
@@ -181,7 +182,7 @@
 static void destroy_connection(int vid);
 static void vhost_guest_notified(int vid);
 
-static const struct vhost_device_ops virtio_net_device_ops =
+static const struct rte_vhost_device_ops virtio_net_device_ops =
 {
     .new_device =  new_device,
     .destroy_device = destroy_device,
@@ -5272,7 +5273,7 @@
                                  struct rte_flow_query_count *query,
                                  struct rte_flow_error *error)
 {
-    struct rte_flow_action_count count = { .shared = 0, .id = 0 };
+    struct rte_flow_action_count count = { .id = 0 };
     const struct rte_flow_action actions[] = {
         {
             .type = RTE_FLOW_ACTION_TYPE_COUNT,
diff -Naur ovs/ofproto/ofproto-dpif.c ovs_tainted/ofproto/ofproto-dpif.c
--- ovs/ofproto/ofproto-dpif.c	2021-11-27 15:39:37.000000000 +0100
+++ ovs_tainted/ofproto/ofproto-dpif.c	2021-12-03 07:49:01.133772620 +0100
@@ -748,7 +748,7 @@
         return 0;
     }
 
-    backer_name = xasprintf("ovs-%s", type);
+    backer_name = xasprintf("%s-%s", getenv("CLOONIX_NET"), type);
 
     /* Remove any existing datapaths, since we assume we're the only
      * userspace controlling the datapath. */
