diff -Naur dpdk/drivers/net/tap/rte_eth_tap.h dpdk_tainted/drivers/net/tap/rte_eth_tap.h
--- dpdk/drivers/net/tap/rte_eth_tap.h	2019-05-25 18:48:13.000000000 +0200
+++ dpdk_tainted/drivers/net/tap/rte_eth_tap.h	2019-05-28 14:52:48.381979132 +0200
@@ -18,11 +18,11 @@
 #include <rte_gso.h>
 #include "tap_log.h"
 
-#ifdef IFF_MULTI_QUEUE
-#define RTE_PMD_TAP_MAX_QUEUES	TAP_MAX_QUEUES
-#else
+// #ifdef IFF_MULTI_QUEUE
+// #define RTE_PMD_TAP_MAX_QUEUES	TAP_MAX_QUEUES
+// #else
 #define RTE_PMD_TAP_MAX_QUEUES	1
-#endif
+// #endif
 #define MAX_GSO_MBUFS 64
 
 enum rte_tuntap_type {
diff -Naur dpdk/lib/librte_eal/linux/eal/eal.c dpdk_tainted/lib/librte_eal/linux/eal/eal.c
--- dpdk/lib/librte_eal/linux/eal/eal.c	2019-05-25 18:48:13.000000000 +0200
+++ dpdk_tainted/lib/librte_eal/linux/eal/eal.c	2019-05-28 14:51:27.653977216 +0200
@@ -109,13 +109,13 @@
 	char tmp[PATH_MAX];
 	int ret;
 
-	if (getuid() != 0) {
+//cloonix	if (getuid() != 0) {
 		/* try XDG path first, fall back to /tmp */
 		if (xdg_runtime_dir != NULL)
 			directory = xdg_runtime_dir;
 		else
 			directory = fallback;
-	}
+//cloonix	}
 	/* create DPDK subdirectory under runtime dir */
 	ret = snprintf(tmp, sizeof(tmp), "%s/dpdk", directory);
 	if (ret < 0 || ret == sizeof(tmp)) {
