diff -Naur openvswitch-2.11.1/tests/dpdk/ring_client.c openvswitch_tainted/tests/dpdk/ring_client.c
--- openvswitch-2.11.1/tests/dpdk/ring_client.c	2019-02-21 06:26:00.525526784 +0100
+++ openvswitch_tainted/tests/dpdk/ring_client.c	2019-04-25 18:03:02.861077852 +0200
@@ -138,6 +138,12 @@
     return 0;
 }
 
+
+static void my_process_bulk(unsigned nb, void *pkts)
+{
+  printf("TODO %u\n", nb);
+}
+
 /*
  * Application main function - loops through
  * receiving and processing packets. Never returns
@@ -191,10 +197,7 @@
         }
 
         if (rx_pkts > 0) {
-            /* blocking enqueue */
-            do {
-                rslt = rte_ring_enqueue_bulk(tx_ring, pkts, rx_pkts, NULL);
-            } while (rslt == -ENOBUFS);
+            my_process_bulk(rx_pkts, pkts);
         }
     }
 }
