diff -Naur openvswitch-2.11.1/lib/netdev-dpdk.c openvswitch_tainted/lib/netdev-dpdk.c
--- openvswitch-2.11.1/lib/netdev-dpdk.c	2019-04-11 04:37:57.481536795 +0200
+++ openvswitch_tainted/lib/netdev-dpdk.c	2019-05-01 18:56:07.477587691 +0200
@@ -5004,6 +5016,7 @@
     .send = netdev_dpdk_eth_send,
 };
 
+//cloonix
 static const struct netdev_class dpdk_ring_class = {
     .type = "dpdkr",
     NETDEV_DPDK_CLASS_BASE,
