diff -Naur ovs/include/openvswitch/vlog.h ovs_tainted/include/openvswitch/vlog.h
--- ovs/include/openvswitch/vlog.h	2019-05-21 21:24:46.000000000 +0200
+++ ovs_tainted/include/openvswitch/vlog.h	2019-05-25 17:53:18.123684656 +0200
@@ -178,7 +178,7 @@
         static struct vlog_module this_module = {                       \
             OVS_LIST_INITIALIZER(&this_module.list),                    \
             #MODULE,                                        /* name */  \
-            { VLL_INFO, VLL_INFO, VLL_INFO },             /* levels */  \
+            { VLL_OFF, VLL_OFF, VLL_INFO },             /* levels */  \
             VLL_INFO,                                  /* min_level */  \
             true                               /* honor_rate_limits */  \
         };                                                              \
diff -Naur ovs/lib/netdev-dpdk.c ovs_tainted/lib/netdev-dpdk.c
--- ovs/lib/netdev-dpdk.c	2019-05-21 21:24:46.000000000 +0200
+++ ovs_tainted/lib/netdev-dpdk.c	2019-05-25 17:53:18.123684656 +0200
@@ -732,6 +732,7 @@
 {
     struct dpdk_mp *dmp, *next;
     bool reuse = false;
+    int ret;
 
     ovs_mutex_lock(&dpdk_mp_mutex);
     /* Check if shared memory is being used, if so check existing mempools
@@ -2030,8 +2031,8 @@
 {
     uint32_t pkt_len = rte_pktmbuf_pkt_len(pkt) - sizeof(struct ether_hdr);
 
-    return rte_meter_srtcm_color_blind_check(meter, profile, time, pkt_len) ==
-                                             e_RTE_METER_GREEN;
+    return rte_meter_srtcm_color_blind_check(meter, profile, time, pkt_len) == 0;
+                                             //e_RTE_METER_GREEN;
 }
 
 static int
@@ -3446,18 +3447,18 @@
         }
     }
 
-    if (VLOG_IS_DBG_ENABLED()) {
-        struct ds mapping = DS_EMPTY_INITIALIZER;
-
-        ds_put_format(&mapping, "TX queue mapping for port '%s':\n",
-                      netdev_get_name(&dev->up));
-        for (i = 0; i < total_txqs; i++) {
-            ds_put_format(&mapping, "%2d --> %2d\n", i, dev->tx_q[i].map);
-        }
-
-        VLOG_DBG("%s", ds_cstr(&mapping));
-        ds_destroy(&mapping);
-    }
+//    if (VLOG_IS_DBG_ENABLED()) {
+//        struct ds mapping = DS_EMPTY_INITIALIZER;
+//
+//        ds_put_format(&mapping, "TX queue mapping for port '%s':\n",
+//                      netdev_get_name(&dev->up));
+//        for (i = 0; i < total_txqs; i++) {
+//            ds_put_format(&mapping, "%2d --> %2d\n", i, dev->tx_q[i].map);
+//        }
+//
+//        VLOG_DBG("%s", ds_cstr(&mapping));
+//        ds_destroy(&mapping);
+//    }
 
     free(enabled_queues);
 }
diff -Naur ovs/lib/vlog.c ovs_tainted/lib/vlog.c
--- ovs/lib/vlog.c	2019-05-21 21:24:46.000000000 +0200
+++ ovs_tainted/lib/vlog.c	2019-05-25 17:53:18.123684656 +0200
@@ -572,7 +572,7 @@
             ovs_fatal(0, "processing \"%s\": %s", arg, msg);
         }
     } else {
-        vlog_set_levels(NULL, VLF_ANY_DESTINATION, VLL_DBG);
+        vlog_set_levels(NULL, VLF_ANY_DESTINATION, VLL_WARN);
     }
 }
 
diff -Naur ovs/ofproto/ofproto-dpif-xlate.c ovs_tainted/ofproto/ofproto-dpif-xlate.c
--- ovs/ofproto/ofproto-dpif-xlate.c	2019-05-21 21:24:46.000000000 +0200
+++ ovs_tainted/ofproto/ofproto-dpif-xlate.c	2019-05-25 17:53:18.127684673 +0200
@@ -4049,6 +4049,12 @@
            xlate_report_error(ctx, "Cannot truncate output to patch port");
        }
        patch_port_output(ctx, xport, xport->peer);
+
+    if (mbridge_has_mirrors(ctx->xbridge->mbridge) && xport->xbundle) {
+        mirror_packet(ctx, xport->xbundle,
+                      xbundle_mirror_dst(xport->xbundle->xbridge,
+                                         xport->xbundle));
+    }
        return;
     }
 
