diff -Naur usbredir/tools/usbredirect.c new_usbredir/tools/usbredirect.c
--- usbredir/tools/usbredirect.c	2021-04-30 18:00:00.000000000 +0200
+++ new_usbredir/tools/usbredirect.c	2021-05-01 10:05:05.534906487 +0200
@@ -528,11 +528,19 @@
     /* Only allow libusb logging if log verbosity is uredirparser_debug_data
      * (or higher), otherwise we disable it here while keeping usbredir's logs enable. */
     if  (self->verbosity < usbredirparser_debug_data)  {
+
+// begin cloonix patch for opensuse153
+#if LIBUSB_API_VERSION >= 0x01000106
         int ret = libusb_set_option(NULL, LIBUSB_OPTION_LOG_LEVEL, LIBUSB_LOG_LEVEL_NONE);
         if (ret != LIBUSB_SUCCESS) {
             g_warning("error disabling libusb log level: %s", libusb_error_name(ret));
             goto end;
         }
+#else
+     libusb_set_debug(NULL, LIBUSB_LOG_LEVEL_NONE);
+#endif
+// end cloonix patch for opensuse153
+
     }
 
     if (self->is_client) {
