diff -Naur virglrenderer/src/vrend_renderer.c new_virglrenderer/src/vrend_renderer.c
--- virglrenderer/src/vrend_renderer.c	2020-06-04 15:11:38.000000000 +0200
+++ new_virglrenderer/src/vrend_renderer.c	2020-06-04 15:40:54.859608469 +0200
@@ -3798,9 +3798,9 @@
          glViewport(ctx->sub->vps[idx].cur_x, cy, ctx->sub->vps[idx].width, ctx->sub->vps[idx].height);
 
       if (idx && has_feature(feat_viewport_array))
-         if (vrend_state.use_gles) {
-            glDepthRangeIndexedfOES(idx, ctx->sub->vps[idx].near_val, ctx->sub->vps[idx].far_val);
-         } else
+//         if (vrend_state.use_gles) {
+//            glDepthRangeIndexedfOES(idx, ctx->sub->vps[idx].near_val, ctx->sub->vps[idx].far_val);
+//         } else
             glDepthRangeIndexed(idx, ctx->sub->vps[idx].near_val, ctx->sub->vps[idx].far_val);
       else
          if (vrend_state.use_gles)
