diff -Naur virglrenderer/src/vrend_renderer.c newvirglrenderer/src/vrend_renderer.c
--- virglrenderer/src/vrend_renderer.c	2019-12-14 20:47:16.000000000 +0100
+++ newvirglrenderer/src/vrend_renderer.c	2019-12-14 21:57:47.562009430 +0100
@@ -3725,9 +3725,9 @@
          glViewport(ctx->sub->vps[idx].cur_x, cy, ctx->sub->vps[idx].width, ctx->sub->vps[idx].height);
 
       if (idx && has_feature(feat_viewport_array))
-         if (vrend_state.use_gles) {
-            glDepthRangeIndexedfOES(idx, ctx->sub->vps[idx].near_val, ctx->sub->vps[idx].far_val);
-         } else
+//         if (vrend_state.use_gles) {
+//            glDepthRangeIndexedfOES(idx, ctx->sub->vps[idx].near_val, ctx->sub->vps[idx].far_val);
+//         } else
             glDepthRangeIndexed(idx, ctx->sub->vps[idx].near_val, ctx->sub->vps[idx].far_val);
       else
          if (vrend_state.use_gles)
