diff -Naur virglrenderer/src/vrend_renderer.c patched_virglrenderer/src/vrend_renderer.c
--- virglrenderer/src/vrend_renderer.c	2019-05-25 18:52:24.000000000 +0200
+++ patched_virglrenderer/src/vrend_renderer.c	2019-05-25 18:53:26.966545537 +0200
@@ -2389,9 +2389,9 @@
          ctx->sub->vps[idx].far_val = far_val;
 
          if (idx && has_feature(feat_viewport_array))
-            if (vrend_state.use_gles) {
-               glDepthRangeIndexedfOES(idx, ctx->sub->vps[idx].near_val, ctx->sub->vps[idx].far_val);
-            } else
+//            if (vrend_state.use_gles) {
+//               glDepthRangeIndexedfOES(idx, ctx->sub->vps[idx].near_val, ctx->sub->vps[idx].far_val);
+//            } else
                glDepthRangeIndexed(idx, ctx->sub->vps[idx].near_val, ctx->sub->vps[idx].far_val);
          else
             if (vrend_state.use_gles)
