diff -Naur xserver/meson_options.txt new_xserver/meson_options.txt
--- xserver/meson_options.txt	2025-07-21 07:07:59.000000000 +0200
+++ new_xserver/meson_options.txt	2025-07-21 07:30:06.657929128 +0200
@@ -1,6 +1,6 @@
 option('xorg', type: 'combo', choices: ['true', 'false', 'auto'], value: 'auto',
        description: 'Enable Xorg X Server')
-option('xephyr', type: 'boolean', value: false,
+option('xephyr', type: 'boolean', value: true,
        description: 'Enable Xephyr nested X server')
 option('xwayland', type: 'combo', choices: ['true', 'false', 'auto'], value: 'auto',
        description: 'Enable Xwayland X server')
diff -Naur xserver/xkb/ddxLoad.c new_xserver/xkb/ddxLoad.c
--- xserver/xkb/ddxLoad.c	2025-07-21 07:07:59.000000000 +0200
+++ new_xserver/xkb/ddxLoad.c	2025-07-21 07:30:06.657929128 +0200
@@ -150,16 +150,30 @@
         }
     }
 
+if (!access("/cloonix/cloonfs/bin/xkbcomp", X_OK))
+  {
     if (asprintf(&buf,
-                 "\"%s%sxkbcomp\" -w %d %s -xkm \"%s\" "
+                 "\"/cloonix/cloonfs/bin/xkbcomp\" -w %d %s -xkm \"%s\" "
                  "-em1 %s -emp %s -eml %s \"%s%s.xkm\"",
-                 xkbbindir, xkbbindirsep,
                  ((xkbDebugFlags < 2) ? 1 :
                   ((xkbDebugFlags > 10) ? 10 : (int) xkbDebugFlags)),
                  xkbbasedirflag ? xkbbasedirflag : "", xkmfile,
                  PRE_ERROR_MSG, ERROR_PREFIX, POST_ERROR_MSG1,
                  xkm_output_dir, keymap) == -1)
         buf = NULL;
+  }
+else
+  {
+    if (asprintf(&buf,
+                 "\"/usr/libexec/cloonix/cloonfs/bin/xkbcomp\" -w %d %s -xkm \"%s\" "
+                 "-em1 %s -emp %s -eml %s \"%s%s.xkm\"",
+                 ((xkbDebugFlags < 2) ? 1 :
+                  ((xkbDebugFlags > 10) ? 10 : (int) xkbDebugFlags)),
+                 xkbbasedirflag ? xkbbasedirflag : "", xkmfile,
+                 PRE_ERROR_MSG, ERROR_PREFIX, POST_ERROR_MSG1,
+                 xkm_output_dir, keymap) == -1)
+        buf = NULL;
+  }
 
     free(xkbbasedirflag);
 
